declarative-examples/simple-examples/whenEnvTrue.groovy

```diff
@@ -0,0 +1,25 @@
+pipeline {
+  agent any
+
+  environment {
+      FOO = "BAR"
+  }
+
+  stages {
+    stage("Hello") {
+      steps {
+        echo "Hello"
+      }
+    }
+    stage("Evaluate FOO") {
+      when {
+        // stage won't be skipped as long as FOO == BAR
+        environment name: "FOO", value: "BAR"
```

Comment:
That's what I get for stealing test cases from @abayer. 

I deleted whenEnvFalse.groovy and changed this to whenEnvVarCondition.groovy with a dynamic environment variable.